searchsummary/src/vespa/searchsummary/docsummary/positionsdfw.cpp

```diff
@@ -113,6 +115,63 @@ PositionsDFW::PositionsDFW(const vespalib::string & attrName) :
 }
 
 namespace {
+
+void
+insertPos(int64_t docxy, vespalib::slime::Inserter &target)
+{
+
+    int32_t docx = 0;
+    int32_t docy = 0;
+    vespalib::geo::ZCurve::decode(docxy, &docx, &docy);
+    if (docx == 0 && docy == INT_MIN) {
+        LOG(spam, "skipping empty zcurve value");
+        return;
+    }
+    vespalib::slime::Cursor &obj = target.insertObject();
+    obj.setLong("y", docy);
+    obj.setLong("x", docx);
+
+    double degrees_ns = docy;
+    degrees_ns /= 1000000.0;
+    double degrees_ew = docx;
+    degrees_ew /= 1000000.0;
+
+    vespalib::asciistream latlong;
+    latlong << vespalib::FloatSpec::fixed;
+    if (degrees_ns < 0) {
+        latlong << "S" << (-degrees_ns);
+    } else {
+        latlong << "N" << degrees_ns;
+    }
+    latlong << ";";
+    if (degrees_ew < 0) {
+        latlong << "W" << (-degrees_ew);
+    } else {
+        latlong << "E" << degrees_ew;
+    }
+    obj.setString("latlong", vespalib::Memory(latlong.str()));
+}
+
+void
+insertFromAttr(const attribute::IAttributeVector &attribute, uint32_t docid, vespalib::slime::Inserter &target)
+{
+    std::vector<IAttributeVector::largeint_t> pos(16);
+    uint32_t numValues = attribute.get(docid, &pos[0], pos.size());
```

Comment:
I suggest using IntegerContent::fill() (searchcommon/src/vespa/searchcommon/attribute/attributecontent.h) instead as that handles resizing and re-trying if necessary.

Reply:
done.